golang/cosmos/x/vstorage/keeper/keeper.go

```diff
@@ -115,10 +119,14 @@ func (k Keeper) ExportStorage(ctx sdk.Context) []*types.DataEntry {
 	defer iterator.Close()
 	for ; iterator.Valid(); iterator.Next() {
 		path := types.EncodedKeyToPath(iterator.Key())
-		value := string(bytes.TrimPrefix(iterator.Value(), types.EncodedDataPrefix))
-		if len(value) == 0 {
+		rawValue := iterator.Value()
+		if len(rawValue) == 0 {
 			continue
 		}
+		if len(rawValue) == 1 && rawValue[0] == types.EncodedNoDataValue[0] {
+			continue
+		}
+		value := string(bytes.TrimPrefix(rawValue, types.EncodedDataPrefix))
```

Comment:
Let's tighten this up:
```suggestion
		if bytes.Equal(rawValue, types.EncodedNoDataValue) {
			continue
		}
		if !bytes.HasPrefix(rawValue, types.EncodedDataPrefix) {
			panic(...)
		}
		value := string(bytes.TrimPrefix(rawValue, types.EncodedDataPrefix))
```

Reply:
`panic` here would technically be a behavioral change, right? I wasn't sure if for some reason there could be data without an encodedDataPrefix. However, I don't see any way the code would introduce those.

Reply:
`value := string(bytes.TrimPrefix(rawValue, types.EncodedDataPrefix))`

I'm not very familiar with go, but why not a slice since we already validated that the prefix is there? I see go 1.20 has a new [`bytes.CutPrefix`](https://pkg.go.dev/bytes#CutPrefix) that [does it with a slice](https://cs.opensource.google/go/go/+/refs/tags/go1.20:src/bytes/bytes.go;l=1367).

Reply:
> `panic` here would technically be a behavioral change

Correct.  But it's a good one!

> why not a slice

Because I'm not very familiar with go. :) Your suggestion is what I was reaching for.